spec/models/piece_spec.rb

```diff
@@ -4,13 +4,140 @@
+      white_pawn = FactoryGirl.create(:pawn, :white, game_id: game.id)
+
+      expect(white_pawn.capturable?(4, 5)).to eq false
+    end
+
+    it 'return false if specified coordinates occupied by same color piece' do
+      game = empty_game
+      white_pawn = FactoryGirl.create(
+        :pawn,
+        :white,
+        game_id: game.id,
+        x_position: 3,
+        y_position: 0,
+        captured: false
+      )
+      other_white_pawn = FactoryGirl.create(
+        :pawn,
+        :white,
+        game_id: game.id,
+        x_position: 4,
+        y_position: 1,
+        captured: false
+      )
+
+      expect(other_white_pawn.capturable?(
+               white_pawn.x_position,
+               white_pawn.y_position
+      )).to eq false
+    end
+
+    it 'return true if specified coordinates occupied by target piece' do
+      game = empty_game
+      black_rook = FactoryGirl.create(
+        :rook,
+        :black,
+        game_id: game.id,
+        x_position: 4,
+        y_position: 0,
+        captured: false
+      )
+      white_pawn = FactoryGirl.create(
+        :pawn,
+        :white,
+        game_id: game.id,
+        x_position: 4,
+        y_position: 1,
+        captured: false
+      )
+
+      expect(white_pawn.capturable?(
+               black_rook.x_position,
+               black_rook.y_position
+      )).to eq true
+    end
+  end
+
+  describe 'capture!' do
+    it 'update target piece attributes' do
+      game = empty_game
+      black_rook = FactoryGirl.create(
+        :rook,
+        :black,
+        game_id: game.id,
+        x_position: 3,
+        y_position: 1,
+        captured: false
+      )
+      white_pawn = FactoryGirl.create(
+        :pawn,
+        :white,
+        game_id: game.id,
+        x_position: 4,
+        y_position: 1,
+        captured: false
+      )
+
+      black_rook.capture!(white_pawn.x_position, white_pawn.y_position)
+
+      white_pawn.reload
+
+      expect(white_pawn.x_position).to eq nil
+      expect(white_pawn.y_position).to eq nil
+      expect(white_pawn.captured).to eq true
+    end
+  end
+
+  describe 'friendly_piece?' do
+    it 'return true if target piece have the same color' do
+      game = empty_game
+      white_pawn = FactoryGirl.create(
+        :pawn,
+        :white,
+        game_id: game.id,
+        x_position: 3,
+        y_position: 0,
+        captured: false
+      )
+      other_white_pawn = FactoryGirl.create(
+        :pawn,
+        :white,
+        game_id: game.id,
+        x_position: 4,
+        y_position: 1,
+        captured: false
+      )
+
+      expect(other_white_pawn.friendly_piece?(
+               white_pawn.x_position,
+               white_pawn.y_position
+      )).to eq true
+    end
+  end
+
+  # before create a new game, destory all pieces pre-populated from game model
+  def empty_game
+    game = FactoryGirl.create(:game)
+    game.pieces.destroy_all
+    game.reload
+    game
+  end
```

Comment:
I normally agree with explicit and readability over DRY, and totally agree with your thought process here @Se7enB2st 

Tests kind of break this rule in my eyes. There is a lot of code that is simply required for setup, which creates the need to break some best practices in the interest of DRY.

Let's take this game creation, for example. We are now adding a feature which will require a new argument when creating a new instance. Let's say the feature is whether you want to join the game as black or white, instead of the implicit black/white join logic we currently have.

In the above example, we now have to refactor every game creation in our code. If we used let, we're looking at maybe 1-2 edits. Without the use of edit, we're looking at 1^n edits, where n is the amount of tests we have involving a game.

All this said, design decision and I abstain.